src/main/java/com/salesforce/functions/jvm/sdk/bulk/BulkApi.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (c) 2021, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+package com.salesforce.functions.jvm.sdk.bulk;
+
+import com.salesforce.functions.jvm.sdk.bulk.builder.JobBuilder;
+import com.salesforce.functions.jvm.sdk.bulk.error.BulkApiException;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+
+/** Bulk API client to upload data in bulk to a Salesforce org. */
+public interface BulkApi {
+  /**
+   * Creates a job representing a bulk operation and its associated data that is sent to Salesforce
+   * for asynchronous processing. Provide job data via {@link #uploadJobData(String, File)}.
+   *
+   * @param job The job to create.
+   * @return The ID of the created job.
+   * @throws BulkApiException If an error occurred during job creation.
+   * @see #newJobBuilder(String, Operation)
+   */
+  @SuppressWarnings("unused")
+  String createJob(Job job) throws BulkApiException;
```

Comment:
The response body for [creating a job](https://developer.salesforce.com/docs/atlas.en-us.api_asynch.meta/api_asynch/create_job.htm) contains more than just an `id`.  Should this method return a richer object instead?

Reply:
I wanted to have a simple identifier as the return value from `createJob` that can be directly passed into `getJobInfo`, `uploadJobData` and the like. If this would return `JobInfo` (I only skimmed the returned structure, but it's remarkably similar to `JobInfo`), would we change `uploadJobData` to accept a `JobInfo` as the identifier for the job? This would mean we'd have to ignore the additional data in that object - why pass it into there then?

We could return `JobInfo` and expect users to call `getId()` on it when using the other methods. I'm not a huge fan of that but not a hill I'm willing to die on. Would you prefer the explicit `getId()` call?

Reply:
> Would you prefer the explicit `getId()` call?

I think I would. In most scenarios where I'm using an API client library to create a resource, I expect to get all the resource data back. It would be weird if you wanted some data out of the job, but would have to make two api calls to get it: `id = createJob(job); del = getJobInfo(id).columnDelimiter`.

Reply:
I checked, we should be able to re-use `JobInfo` for this - added in [6865ae7](https://github.com/forcedotcom/sf-fx-sdk-java/pull/64/commits/6865ae7c5981059306e5de9a8e2a0feab0adcc09).